docs/data/data-grid/pivoting/pivoting.md

```diff
@@ -2,16 +2,175 @@
+
+{{"demo": "GridPivotingMultipleValues.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Getting started with pivoting
+
+### Pivot model
+
+The pivot model is a configuration object that defines rows, columns, and values of the pivot data grid:
+
+```tsx
+interface GridPivotModel {
+  rows: Array<{
+    field: string;
+    hidden?: boolean;
+  }>;
+  columns: Array<{
+    field: string;
+    sort?: 'asc' | 'desc';
+    hidden?: boolean;
+  }>;
+  values: Array<{
+    field: string;
+    aggFunc: string;
+    hidden?: boolean;
+  }>;
+}
+```
+
+### Initialize pivoting
+
+The easiest way to initialize pivoting is with an uncontrolled mode using the `initialState` prop.
+You can initialize the pivot model, pivot panel visibility, and enable the pivot mode:
+
+```tsx
+<DataGridPremium
+  initialState={{ pivoting: { model: pivotModel, enabled: true, panelOpen: true } }}
+/>
+```
+
+{{"demo": "GridPivotingInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Controlled pivoting
+
+For fully controlled pivoting state, there are the following controlled props:
+
+- Pivot model:
+  - `pivotModel`: Controls the current pivot configuration.
+  - `onPivotModelChange`: Callback fired when the pivot model changes.
+- Pivot mode toggle:
+  - `pivotActive`: Controls whether pivot mode is active.
+  - `onPivotActiveChange`: Callback fired when pivot mode is active/inactive.
+- Pivot panel:
+  - `pivotPanelOpen`: Controls whether the pivot panel is open.
+  - `onPivotPanelOpenChange`: Callback fired when the pivot panel is opened/closed.
+
+{{"demo": "GridPivotingControlled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Using fields in the pivot model multiple times
+
+While this is not supported yet, we are working to bring this feature to the Data Grid.
+
+## Disable pivoting
+
+To disable the pivoting feature, set the `disablePivoting` prop to `true`:
+
+```tsx
+<DataGridPremium disablePivoting />
+```
+
+## Derived columns in pivot mode
+
+In pivot mode, it's often useful to group data by a year or quarter.
+Data Grid automatically generates year and quarter columns for each `date` column.
+
+In the demo below, the `transactionDate` column is represented by additional columns in the pivot mode: `transactionDate-year` and `transactionDate-quarter`:
+
+{{"demo": "GridPivotingFinancial.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom derived columns
+
+To customize the derived columns, use the `getPivotDerivedColumns` prop.
+It is called for each original column and should return an array of derived columns, or `undefined` if no derived columns are needed.
+
+{{"demo": "GridGetPivotDerivedColumns.js", "bg": "inline", "defaultCodeOpen": true}}
+
+## Sticky column groups
+
+Depending on the pivot mode, some column groups might exceed the width of the Data Grid viewport.
+To improve the user experience, you can make these column groups "sticky" so that the column group labels are visible while scrolling horizontally.
```

Comment:
```suggestion
To improve the user experience, you can make these column groups "sticky" so that the column group labels remain visible while scrolling horizontally.
```